Part-2: Compiler Warnings as Errors - WordPress Module - FragmentManager

07jasjeet opened the pull request:
# ONLY merge this after [Part 1](https://github.com/wordpress-mobile/WordPress-Android/pull/19471) has been merged! (Derivative branch)

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/17253 partially. (Sub-ticket of https://github.com/wordpress-mobile/WordPress-Android/pull/17246)

To Test:

- [AddCategoryFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java): 

   1) Have a site with a post.
   2) Navigate to "My Site" and click on "Posts" to see all your posts. Open any post in edit mode.
   3) Click on dropdown button on top-left.
   4) Click "Post settings".
   5) Select/Open "Categories".
   6) Click on "+" symbol on top left.
   7) A dialog should open without errors.
- [CollapseFullScreenDialogFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/CollapseFullScreenDialogFragment.java):

   1. Navigate to "Reader" tab.
   2. On any post, click on "Comment" to open comments.
   3. On the bottom of the screen, you will have the text field to reply. Beside that, you will have an "Arrow up" button. Click it.
   4. This is our full screen dialog that got changed. It should work as expected.
- [SignupEpilogueFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java):

   1. Just follow the **signup** flow till you reach a form where you have to fill your name, display name and password.
   2. This is hard to test (and I need help here). The changes in this file are minute and are certainly not going to break anything afaik.

- [SitePickerActivity.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java):

  1. Navigate to "My Site" tab.
  2. Click on "Add Site" button if you have no sites else, click on current opened site's title on top of the page (with an arrow).
  3. You will have "Choose site" screen infront of you.
  4. Click on "+" button on top right. (on Action bar)
  5. Dialog should work as expected.

- [PersonDetailFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/people/PersonDetailFragment.java):

  1. Have at least one site.
  2. Navigate to "My Site" tab.
  3. Click on "More".
  4. Click on "People"
  5. Click on any person.
  6. A detailed view of the person should pop up as expected.

## Regression Notes
1. Potential unintended areas of impact
   Mostly none because the newer API's are a direct replacement.

7. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual Testing

8. What automated tests I added (or what prevented me from doing so)
   None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

07jasjeet committed:
Update AddCategoryFragment.java

07jasjeet committed:
Update FullScreenDialogFragment.java

07jasjeet committed:
Update DomainRegistrationDetailsFragment.kt

07jasjeet committed:
Update SignupEpilogueFragment.java

07jasjeet committed:
Update CollapseFullScreenDialogFragment.java

07jasjeet committed:
Update SitePickerActivity.java

07jasjeet committed:
Update PersonDetailFragment.java

07jasjeet commented:
@antonis Please add yourself as a reviewer.

antonis commented on WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java:
```diff
@@ -336,7 +336,7 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
             }
         } else {
-            mDialog = (FullScreenDialogFragment) getFragmentManager().findFragmentByTag(FullScreenDialogFragment.TAG);
+            mDialog = (FullScreenDialogFragment) getParentFragmentManager().findFragmentByTag(FullScreenDialogFragment.TAG);
```
I've noticed a minor checkstyle issue on this line:
```
Line is longer than 120 characters (found 124). [LineLength]
```

07jasjeet replied:
Oops looks like the checkstyle error might have went out of screen as well haha. Will fix it!

07jasjeet replied:
Also i will be leaving test directions soon so I would request you to wait on this one until then.

07jasjeet replied:
@antonis test instructions are up!

antonis replied:
Thank you for the heads up @07jasjeet 🙇 
My plan is to review the PR later today or tomorrow.

07jasjeet committed:
Merge remote-tracking branch 'upstream/trunk' into 17253-compiler-warnings-part-2

07jasjeet committed:
Replaced NullPointerExpection inducing APIs

Needs no testing. The behaviour should be same as the new API's will throw IllegalStateException instead of NullPointerException if activity was null.

07jasjeet committed:
Documentation updated with hyperlink

07jasjeet committed:
Checkstyle error resolved

antonis commented on WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java:
```diff
@@ -833,36 +833,36 @@ public static void addSite(Activity activity, boolean hasAccessToken, SiteCreati
         @Override
-        @SuppressWarnings("deprecation")
         public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+            assert getArguments() != null;
```
Thank you for adding this extra check @07jasjeet 👍 
What would you say on replacing it with `requireArguments();`

(thank you @ParaskP7 for the suggestion 🙇)

antonis commented on WordPress/src/main/java/org/wordpress/android/ui/people/PersonDetailFragment.java:
```diff
@@ -83,9 +83,10 @@ public static PersonDetailFragment newInstance(long currentUserId, long personId
+        ((WordPress) requireActivity().getApplicationContext()).component().inject(this);
 
         if (savedInstanceState == null) {
+            assert getArguments() != null;
```
As above https://github.com/wordpress-mobile/WordPress-Android/pull/19490/files#r1384759546

07jasjeet committed:
Requested changes

1) Used requireArguments() instead of assertion.

antonis reviewed (approved):
Thank you for your work on this @07jasjeet and for handling the review feedback 🙇 
I've tested the implementation and everything worked as expected for me:
- [x] AddCategoryFragment.java
- [x] CollapseFullScreenDialogFragment.java
- [x] SignupEpilogueFragment.java
- [x] SitePickerActivity.java
- [x] PersonDetailFragment.java
- [x] FullScreenDialogFragment.java

The code also looks good 🎉
